src/codegen/abi.js

```diff
@@ -54,28 +56,15 @@ module.exports = class AbiCodeGenerator {
         )
 
         event.get('inputs').forEach((input, index) => {
-          let name = input.get('name')
-          let inputType = input.get('indexed')
-            ? this._indexedInputType(input.get('type'))
-            : input.get('type')
-
-          if (name === undefined || name === null || name === '') {
-            name = `param${index}`
-          }
-
-          paramsClass.addMethod(
-            tsCodegen.method(
-              `get ${name}`,
-              [],
-              typesCodegen.ascTypeForEthereum(inputType),
-              `
-            return ${typesCodegen.ethereumValueToAsc(
-              `this._event.parameters[${index}].value`,
-              inputType,
-            )}
-            `,
-            ),
+          // Generate getters and classes for event params
+          let paramObject = this._generateEventParam(
+            input,
+            index,
+            eventClassName,
+            `event`,
           )
+          paramsClass.addMethod(paramObject.getter)
+          tupleClasses.push(...paramObject.classes)
```

Comment:
Hehe, I guess this could also just be `tupleClasses.concat(paramObject.classes)`.

Reply:
This allows us to change it in place though and not create a new reference :).

Reply:
Ah, good point!